openff/interchange/components/interchange.py

```diff
@@ -108,17 +109,20 @@ def _check_supported_handlers(cls, force_field: ForceField):
 
         unsupported = list()
 
-        for handler in force_field.registered_parameter_handlers:
-            if handler in {"ToolkitAM1BCC"}:
+        for handler_name in force_field.registered_parameter_handlers:
+            if handler_name in {"ToolkitAM1BCC"}:
                 continue
-            if handler not in _SUPPORTED_SMIRNOFF_HANDLERS:
-                unsupported.append(handler)
+            if handler_name not in _SUPPORTED_SMIRNOFF_HANDLERS:
+                unsupported.append(handler_name)
+            if handler_name in {"Bonds", "ProperTorsions"}:
+                handler = force_field[handler_name]
+                if any([p.k_bondorder for p in handler.parameters]):
+                    raise SMIRNOFFHandlersNotImplementedError(
+                        "Bond order interpolation not supported. Found parameters with attribute"
+                        "`k_bondorder` in handler {handler_name}"
+                    )
```

Comment:
I think this is a good idea! I wonder if rather than hard coding the exact unsupported attributes here, we could instead add a `supported_attributes` class method to each potential handler which explicitly states those that are currently supported, and then generalize the check here to make use of those? 

Having this info may also be useful when performing other validations!

Reply:
That does sound like a better approach - I've implemented it and it passes all tests!